File: BodyJournalAPI/Entities/Exercise.cs
namespace BodyJournalAPI.Entities
{
  public class Exercise
  {
    public int Id { get; set; }
    public int Weight { get; set; }
    public int Reps { get; set; }
    public int Sets { get; set; }
    public string Name { get; set; }
    public string MuscleGroup { get; set; }
    public int Intensity { get; set; }
  }
}

File: app/src/views/Exercises/Exercise.js
import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
const Exercise = ({ match, location }) => {
  const {
    params: { id },
  } = match;

  const exercise = useSelector((state) => state.exercise);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(exerciseActions.getExercise(id));
  }, []);
  return (
    <React.Fragment>
      <p>{exercise.name}</p>
      <p>{exercise.weight}</p>
      <p>{exercise.reps}</p>
      <p>{exercise.sets}</p>
      <p>{exercise.intensity}</p>
    </React.Fragment>
  );
};
export { Exercise };
